plugins/modules/ipaconfig.py

```diff
@@ -0,0 +1,441 @@
+        required: false
+        aliases: ["searchrecordslimit"]
+    ipausersearchfields:
+        description:
+        - Set comma-separated list of fields to search for user search
+        required: false
+        aliases: ["usersearch"]
+    ipagroupsearchfields:
+        description:
+        - Set comma-separated list of fields to search for group search
+        required: false
+        aliases: ["groupsearch"]
+    ipamigrationenabled:
+        description: Enable migration mode
+        type: bool
+        required: false
+        aliases: ["enable-migration"]
+    ipagroupobjectclasses:
+        description: Set default group objectclasses (comma-separated list)
+        required: false
+        type: list
+        aliases: ["groupobjectclasses"]
+    ipauserobjectclasses:
+        description: Set default user objectclasses (comma-separated list)
+        required: false
+        type: list
+        aliases: ["userobjectclasses"]
+    ipapwdexpadvnotify:
+        description:
+        - Set number of days's notice of impending password expiration
+        - values 0 to 2147483647
+        required: false
+        aliases: ["pwdexpnotify"]
+    ipaconfigstring:
+        description: Set extra hashes to generate in password plug-in
+        required: false
+        type: list
+        choices:
+        - "AllowNThash"
+        - "KDC:Disable Last Success"
+        - "KDC:Disable Lockout"
+        - "KDC:Disable Default Preauth for SPNs"
+    ipaselinuxusermaporder:
+        description: Set order in increasing priority of SELinux users
+        required: false
+        type: list
+    ipaselinuxusermapdefault:
+        description: Set default SELinux user when no match found in map rule
+        required: false
+    ipakrbauthzdata:
+        description: set default types of PAC supported for services
+        required: false
+        type: list
+        choices: ["MS-PAC", "PAD", "nfs:NONE"]
+        aliases: ["pac-type"]
+    ipauserauthtype:
+        description: set default types of supported user authentication
+        required: false
+        type: list
+        choices: ["password", "radius", "otp", "disabled"]
+        aliases: ["user-auth_type"]
+    ipadomainresolutionorder:
+        description: set list of domains used for short name qualification
+        required: false
+        type: list
+        aliases: ["domain-resolution_order"]
+'''
+
+EXAMPLES = '''
+---
+- name: Playbook to handle global configuration options
+  hosts: ipaserver
+  become: true
+  tasks:
+    - name: return current values of the global configuration options
+      ipaconfig:
+        ipaadmin_password: password
+      register: result
+    - name: display default login shell
+      debug:
+        msg: '{{result.config.ipadefaultloginshell[0] }}'
+
+    - name: set ipadefaultloginshell and ipamaxusernamelength
+      ipaconfig:
+        ipaadmin_password: password
+        ipadefaultloginshell: /bin/bash
+        ipamaxusernamelength: 64
```

Comment:
The examples need not be a complet YAML file. Please, break the example in two entries, one for setting values, one for retrieving them. See plugins ipahost and ipavault.